File: TopCoder02/Java/InterestingParty.java
import java.util.HashMap;

public class InterestingParty {

	public int bersInvitation(String[] first, String[] second) {
		HashMap<String, Integer> dic = new HashMap<>();
		for(int i = 0; i<first.length;i++ ){
			dic.put(first[i], 0);
			dic.put(second[i], 0);
		}
		for(int i = 0; i<first.length; i++){
			dic.put(first[i], dic.get(first[i])+1);
			dic.put(second[i], dic.get(second[i])+1);
		}
		int ans = 0;
		for(String key : dic.keySet()){
			ans = Math.max(ans, dic.get(key));
		}
		return ans;
	}

}

File: TopCoder02/Scala/InterestingParty.scala
class InterestingParty {
	def bestInvitation(first:Array[String], second:Array[String]):Int = {
	  var dic = Map.empty[String,Int]
	  for(i <- 0 until first.length){
	    dic += first(i) -> 0
	    dic += second(i) -> 0
	  }
	  for(i <- 0 until first.length){
	    dic += first(i) -> (dic(first(i))+1)
	    dic += second(i) -> (dic(second(i))+1)
	  }
	  var ans = 0
	  for(value <- dic.values){
	    ans = math.max(ans, value)
	  }
	  return ans
	}
}